arrayLit not handling func types

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

type fn func(string, string) bool

var funcs = [][]fn{
	{cmpLessFn, nil},
	{cmpGreaterFn, nil},
}

func cmpLessFn(a string, b string) bool {
	return a < b
}

func cmpGreaterFn(a string, b string) bool {
	return a > b
}

func main() {
	for _, fns := range funcs {
		fmt.Println(fns[0]("a", "b"))
	}
}
```


### Expected result

```console
$ go run sample.go
true
false
```


### Got

```console
$ go run ./cmd/yaegi/ run sample.go
sample.go:8:2: panic
run: reflect.Set: value of type func(string, string) bool is not assignable to type *interp.node
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1()
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/program.go:117 +0x94
panic({0x18ce9a0, 0xc000207210})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/traefik/yaegi/interp.runCfg.func1()
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/run.go:193 +0x145
panic({0x18ce9a0, 0xc000207210})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
reflect.Value.assignTo({0x18f4940, 0xc0002083e8, 0xc00020cf40}, {0x19f09e5, 0xb}, 0x19b54a0, 0x0)
	/usr/local/go/src/reflect/value.go:2797 +0x2ad
reflect.Value.Set({0x19b54a0, 0xc000207208, 0x2}, {0x18f4940, 0xc0002083e8, 0xc0004890e0})
	/usr/local/go/src/reflect/value.go:1905 +0xec
github.com/traefik/yaegi/interp.arrayLit.func1(0xc0001aba00)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/run.go:2347 +0x125
github.com/traefik/yaegi/interp.runCfg(0xc00047e240, 0xc00035c000, 0xc00047e240, 0x1)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/run.go:201 +0x2ac
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc00035a000, 0xc000482ea0, 0xc0002119f0)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/run.go:120 +0x385
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0xc00035a000, 0xc00048bb90)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/program.go:140 +0x194
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc00035a000, {0xc0002dadc0, 0x148}, {0x7ffeefbff9ff, 0xc0002dab00}, 0x48)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/interp.go:584 +0x59
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc00035a000, {0x7ffeefbff9ff, 0xf})
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/interp/interp.go:477 +0xab
main.runFile(0x7ffeefbff9ff, {0x7ffeefbff9ff, 0xf}, 0x0)
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:149 +0x8f
main.run({0xc0000320e0, 0x10709e0, 0xc00005c6b0})
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:112 +0xb74
main.main()
	/Users/nrwiersma/Go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:133 +0xd5

exit status 1
```


### Yaegi Version

4653d8729892a84f69ccfc7e58e0d78e71eccdd2

### Additional Notes

_No response_

--- a/_test/composite18.go
+++ b/_test/composite18.go
@@ -1,0 +1,32 @@
+package main
+
+import "fmt"
+
+type fn func(string, string) bool
+
+var funcs = []fn{
+	cmpLessFn,
+	cmpGreaterFn,
+	nil,
+}
+
+func cmpLessFn(a string, b string) bool {
+	return a < b
+}
+
+func cmpGreaterFn(a string, b string) bool {
+	return a > b
+}
+
+func main() {
+	for _, f := range funcs {
+		if f == nil {
+			continue
+		}
+		fmt.Println(f("a", "b"))
+	}
+}
+
+// Output:
+// true
+// false

--- a/_test/composite19.go
+++ b/_test/composite19.go
@@ -1,0 +1,33 @@
+package main
+
+import "fmt"
+
+type fn func(string, string) bool
+
+var funcs = map[string]fn{
+	"less":    cmpLessFn,
+	"greater": cmpGreaterFn,
+	"none":     nil,
+}
+
+func cmpLessFn(a string, b string) bool {
+	return a < b
+}
+
+func cmpGreaterFn(a string, b string) bool {
+	return a > b
+}
+
+func main() {
+	for _, n := range []string{"less", "greater", "none"} {
+		f := funcs[n]
+		if f == nil {
+			continue
+		}
+		fmt.Println(f("a", "b"))
+	}
+}
+
+// Output:
+// true
+// false

--- a/interp/value.go
+++ b/interp/value.go
@@ -215,7 +215,7 @@ func genDestValue(typ *itype, n *node) func(*frame) reflect.Value {
 	switch {
 	case isInterfaceSrc(typ) && !isEmptyInterface(typ):
 		return genValueInterface(n)
-	case isFuncSrc(typ) && n.typ.cat == valueT:
+	case isFuncSrc(typ) && (n.typ.cat == valueT || n.typ.cat == nilT):
 		return genValueNode(n)
 	case typ.cat == valueT && isFuncSrc(n.typ):
 		return genFunctionWrapper(n)

